### DISCUSSION TITLE
While Level.hierarchy will be binary at some point, switching to using UnturnedDat seems wise.

### DISCUSSION THREAD

**Discussion by Jdance-Media** [SmartlyDressedGames/Unturned-3.x-Community#4067] [Ideas]
A lot of the original IFormattedFile stuff was dumped when UnturnedDat came out. The Level.hierarchy appears to be a bunch of .asset files combined. The original interface is somewhat different compared to UnturnedDat, so I'm not quite too sure if this is a difficult idea to execute. 🤷‍♂️

**Comment by SDGNelson** [comment]
Yes, there is quite a bit of mostly-removed-but-slightly-lingering stuff ☹️
Although UnturnedDat is more thoroughly tested, I'm not sure there would really be benefit in replacing for level hierarchy in the meantime. You're right converting to binary is the ideal goal

**Reply by Jdance-Media** [reply]
👍. While I'm at it, thank you for making UnturnedDat a separate library, really helps for making programs for modding.

**Reply by SDGNelson** [reply]
Thanks, I'm glad to hear that was useful! :)
